internal/services/runservice/api/api.go

```diff
@@ -311,6 +311,108 @@ func sendLogs(w http.ResponseWriter, r io.Reader) error {
 	}
 }
 
+type LogsDeleteHandler struct {
+	log *zap.SugaredLogger
+	e   *etcd.Store
+	ost *objectstorage.ObjStorage
+	dm  *datamanager.DataManager
+}
+
+func NewLogsDeleteHandler(logger *zap.Logger, e *etcd.Store, ost *objectstorage.ObjStorage, dm *datamanager.DataManager) *LogsDeleteHandler {
+	return &LogsDeleteHandler{
+		log: logger.Sugar(),
+		e:   e,
+		ost: ost,
+		dm:  dm,
+	}
+}
+
+func (h *LogsDeleteHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
+	q := r.URL.Query()
+
+	runID := q.Get("runid")
+	if runID == "" {
+		http.Error(w, "", http.StatusBadRequest)
+		return
+	}
+	taskID := q.Get("taskid")
+	if taskID == "" {
+		http.Error(w, "", http.StatusBadRequest)
+		return
+	}
+
+	_, setup := q["setup"]
+	stepStr := q.Get("step")
+	if !setup && stepStr == "" {
+		http.Error(w, "", http.StatusBadRequest)
+		return
+	}
+	if setup && stepStr != "" {
+		http.Error(w, "", http.StatusBadRequest)
+		return
+	}
+
+	var step int
+	if stepStr != "" {
+		var err error
+		step, err = strconv.Atoi(stepStr)
+		if err != nil {
+			http.Error(w, "", http.StatusBadRequest)
+			return
+		}
+	}
+
+	if err, deleteError := h.deleteTaskLogs(ctx, runID, taskID, setup, step, w); err != nil {
+		h.log.Errorf("err: %+v", err)
+		if deleteError {
+			switch err.(type) {
+			case common.ErrNotExist:
+				http.Error(w, err.Error(), http.StatusNotFound)
+			default:
+				http.Error(w, err.Error(), http.StatusInternalServerError)
+			}
+		}
+	}
+}
+
+func (h *LogsDeleteHandler) deleteTaskLogs(ctx context.Context, runID, taskID string, setup bool, step int, w http.ResponseWriter) (error, bool) {
+	r, err := store.GetRunEtcdOrOST(ctx, h.e, h.dm, runID)
+	if err != nil {
+		return err, true
+	}
+	if r == nil {
+		return errors.Errorf("no such run with id: %s", runID), true
+	}
+
+	task, ok := r.Tasks[taskID]
+	if !ok {
+		return errors.Errorf("no such task with ID %s in run %s", taskID, runID), true
+	}
+	if len(task.Steps) <= step {
+		return errors.Errorf("no such step for task %s in run %s", taskID, runID), true
+	}
+
+	if task.Steps[step].LogPhase == types.RunTaskFetchPhaseFinished {
+		var logPath string
+		if setup {
+			logPath = store.OSTRunTaskSetupLogPath(task.ID)
+		} else {
+			logPath = store.OSTRunTaskStepLogPath(task.ID, step)
+		}
+		err := h.ost.DeleteObject(logPath)
+		if err != nil {
+			if err == ostypes.ErrNotExist {
+				return common.NewErrNotExist(err), true
+			}
+			return err, true
+		}
+		return nil, false
+	}
+	return errors.Errorf("no logs on OST for for task %s in run %s", taskID, runID), true
```

Comment:
use `util.NewErrBadRequest` and the error should explain that the log is not yet archived.